libs/platform/src/lib/components/carosuel/carosuel.component.ts

```diff
@@ -0,0 +1,526 @@
+                        ? this._currentActiveSlide
+                        : 0;
+                break;
+            case Direction.PREV:
+                nextSlideIndex =
+                    this._currentActiveSlide > 0
+                        ? this._currentActiveSlide - 1
+                        : !force && this.noWrap
+                            ? this._currentActiveSlide
+                            : this.slides.length - 1;
+                break;
+            default:
+                throw new Error('Unknown direction');
+        }
+        return nextSlideIndex;
+    }
+
+    private _select(index: number): void {
+        if (isNaN(index)) {
+            this.pause();
+            return;
+        }
+        const currentSlide = this.slides[this._currentActiveSlide];
+        if (currentSlide) {
+            currentSlide.active = false;
+        }
+        const nextSlide = this.slides[index];
+        if (nextSlide) {
+            this._currentActiveSlide = index;
+            nextSlide.active = true;
+            this.activeSlide = index;
+        }
+        this.cdRef.markForCheck();
+    }
+
+    private restartTimer(): any {
+        this.resetTimer();
+        if (this.isBrowser) {
+            const interval = +this.interval;
+            if (!isNaN(interval) && interval > 0) {
+                this.currentInterval = setInterval(() => {
+                    const nInterval = +this.interval;
+                    if (this.isPlaying && !isNaN(this.interval) && nInterval > 0 && this.slides.length) {
+                        this.nextSlide();
+                    } else {
+                        this.pause();
+                    }
+                }, interval);
+            }
+        }
+    }
+
+    private resetTimer(): void {
+        if (this.isBrowser) {
+            if (this.currentInterval) {
+                clearInterval(this.currentInterval);
+                this.currentInterval = void 0;
+            }
+        }
+    }
+
+    protected hasClass(el: any, className: any) {
+        if (el.classList) {
+            return el.classList.contains(className);
+        } else {
+            return !!el.className.match(new RegExp('(\\s|^)' + className + '(\\s|$)'));
+        }
+    }
+
+    protected classAdd(el: any, className: any) {
+        if (el.classList) {
+            el.classList.add(className);
+        } else if (!this.hasClass(el, className)) {
+            el.className += ' ' + className;
+        }
+    }
+
+    protected removeClass(el: any, className: any) {
+        if (el.classList) {
+            el.classList.remove(className);
+        } else if (this.hasClass(el, className)) {
+            const reg = new RegExp('(\\s|^)' + className + '(\\s|$)');
+            el.className = el.className.replace(reg, ' ');
+        }
+    }
+
+    @HostListener('keyup', ['$event']) keyboardControl(event: KeyboardEvent) {
+        if (this.keyboard) {
+
+            if (event.keyCode === RIGHT_ARROW) {
```

Comment:
It looks like `keyCode` is being deprecated:

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Use `key` property instead:
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

Reply:
`key` is not supported by IE11, I'd rather recommend to use `keyCode` or both.